server/src/main/java/org/apache/druid/guice/DruidBinders.java

```diff
@@ -62,32 +66,68 @@ public static MapBinder<Class<? extends Query>, QueryLogic> queryLogicBinderType
     );
   }
 
-  public static QueryLogicBinder queryLogicBinder(Binder binder)
+  public static QueryBinder queryBinder(Binder binder)
   {
-    return new QueryLogicBinder(binder);
+    return new QueryBinder(binder);
   }
 
-  public static class QueryLogicBinder
+  public static class QueryBinder
```

Comment:
This class is hiding things from people if they ever need them.  I'm not sure if it will ever be needed, but it's relatively easy for this class to expose things that allow someone to do whatever they need while still maintaining the simplicity for the common case.

Reply:
They can do whatever they want by using the `binder` directly.

I don't think they'll miss anything by not knowing about those things - it just does the standard registration.